Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp

```diff
@@ -0,0 +1,253 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Seeding/Seed.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <optional>
+#include <vector>
+
+namespace Acts {
+/// @todo:
+/// 1) Implement the simple Line and Circle fit based on Taubin Circle fit
+/// 2) Implement the simple Line and Parabola fit (from HPS reconstruction by
+/// Robert Johnson)
+
+/// Estimate the track parameters on transverse plane
+///
+/// @note Using method based on V. Karimaki NIM A305 (1991) 187-191 - no weights
+/// are used in Karimaki's fit, d0 is the distance of the closest approach to
+/// the origin, 1/R is the curvature, phi is the angle of the direction
+/// propagation (counter clockwise as positive)
+///
+/// @tparam external_spacepoint_t The type of space point
+/// @param sps is a vector of space points
+///
+/// @return optional estimated parameters d0, 1/R and phi
+template <typename external_spacepoint_t>
+std::optional<std::array<ActsScalar, 3>> estimateTrackParamsFromSeed(
```

Comment:
The estimated track parameters using this method is the (d0, phi, 1/R). I'm not very sure how we could transform them into the meaningful bound parameters.

Reply:
In the general bound parameters vector this can be stored as `eLoc0`, `ePhi`, and `eQOverP`. Unset parameters can be set either to zero or to NaN.

My worry is that the name suggests that it returns track parameters when it does not do so. And you also end up with having magic numbers in the code due to the indices for the array. If you want to return only the three values, I would suggest to define a helper struct so the three variables `d0`, `phi`, and `curvature` can be accessed by name. If you return a `BoundVector` you already get the access by enum for free. The documentation for `eQOverP` also mentions that the name is not always correct anyways (can be q/p or 1/p depending on context); storing a different curvature-like parameter in there would not be too much of a stretch.

Reply:
> My worry is that the name suggests that it returns track parameters when it does not do so. 

P.S. Especially if there exists another function with the same name that returns a different value. I think it is Ok if there are multiple functions with the same name but different input parameters, but they should return the same type for consistency.

Reply:
That makes sense. A bound parameters with `eLoc0`, `ePhi` and `eQOverP` set (others are zero) are returned. I had add some documentation for this.